include/kernel/printer.h

```diff
@@ -23,7 +39,59 @@ class Printer
     static void printlnNumbers(uint64 *numbers, uint8 size);
     static void printlnAddresses(uint64 *numbers, uint8 size);
     static void clearScreen();
+
+  private:
+    /*
+     * update cursor position
+     * @param len number of pixel to be moved in vertical
+     */
+    static void updatePointer(int len);
+    static void printfHelper(int i, const char *format);
+    template <typename T, typename... Args>
+    static void printfHelper(int i, const char *format, T first, Args... args);
 };
 
+template <typename T, typename... Args>
+void Printer::printfHelper(int i, const char *format, T first, Args... args) {
+    char tmp[10] = {0};
+
+    for (; format[i] != '\0'; i++) {
+        if (format[i] == '%' && format[i + 1] != '\0') {
```

Comment:
Sometime, we init a string by using new operator. In this case, the terminated character is not '\0'